libs/aws/odc/aws/__init__.py

```diff
@@ -314,6 +316,20 @@ def s3_open(url: str,
     oo = s3.get_object(Bucket=bucket, Key=key, **kwargs)
     return oo['Body']
 
+def cache_s3_object(url,
```

Comment:
This does not actually cache an object, it will perform new download each time every time to a new path location.

Also is this really functionality we need in this library? I think simple `s3_download` is all that is needed. Let me implement that in the way this library works.